### PR TITLE
feat: clickable speed btns in nav sidebar - dl & ul filters

### LINKED ISSUE #1879: [Feature Request]: Turn the current speed indicators in the sidebar into shortcuts for activating the downloading and uploading torrent state filters
### Description

Whenever I open the sidebar and notice higher-than-usual speed values, I feel compelled to click on them, as they already resemble buttons, to quickly see which torrents are generating that activity. Although I have the state filter in the top bar, it's often combined with other tag, category, or tracker filters, making it not instantly accessible.

### Describe the solution you'd like

What I imagine is tapping or clicking on the upload speed value would temporarily activate the 'uploading' torrent state filter and remove any other enabled filters. Clicking again would remove the 'uploading' filter and restore the previously active filter state. There could be some visual indication when these shortcuts are active, such as a different border or background color.

### Describe alternatives you've considered

_No response_

### Complementary informations

- [ ] Is this feature already implemented in the default WebUI?

### PR DESCRIPTION
**Pull request by jagadam97**
# clickable current speed btns in nav sidebar to apply dl & ul filters [feat]

Closes #1879

I have developed speedCardClickable which can accept clicks to apply filters of torrentState accordingly.
clicking on same button again will remove the filter applied.

proof of testing: https://youtu.be/CWuP3WRYBcw

## PR Checklist

- [x] I've started from master
- [x] I've only committed changes related to this PR
- [x] All tests pass
- [x] I've removed all commented code

### CODE REVIEW

File: src/components/Core/SpeedCardClickable.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<script setup lang="ts">
+import { formatSpeedUnit, formatSpeedValue } from '@/helpers'
+import { useVueTorrentStore } from '@/stores'
+
+defineProps<{
+  icon: string
+  color: string
+  value: number
+  active: boolean
+  handleClick?: () => void
```
Comment (Larsluph) [inline src/components/Core/SpeedCardClickable.vue:10]: Switch function prop to emit

File: src/components/Navbar/SideWidgets/CurrentSpeed.vue [hunk 2]
```diff
@@ -1,12 +1,40 @@
 <script setup lang="ts">
 import SpeedCard from '@/components/Core/SpeedCard.vue'
+import SpeedCardClickable from '@/components/Core/SpeedCardClickable.vue'
 import { useMaindataStore, useVueTorrentStore } from '@/stores'
 import { storeToRefs } from 'pinia'
 import { useI18nUtils } from '@/composables'
+import { useTorrentStore } from '@/stores'
+import { TorrentState } from '@/constants/vuetorrent'
+import { computed } from 'vue'
```
Comment (Larsluph) [inline src/components/Navbar/SideWidgets/CurrentSpeed.vue:9]: Sort and merge imports

```suggestion
import SpeedCard from '@/components/Core/SpeedCard.vue'
import SpeedCardClickable from '@/components/Core/SpeedCardClickable.vue'
import { useI18nUtils } from '@/composables'
import { TorrentState } from '@/constants/vuetorrent'
import { useMaindataStore, useTorrentStore, useVueTorrentStore } from '@/stores'
import { storeToRefs } from 'pinia'
import { computed } from 'vue'
```

File: src/components/Navbar/SideWidgets/CurrentSpeed.vue [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
 
 const { t } = useI18nUtils()
 const { serverState } = storeToRefs(useMaindataStore())
 const { displayGraphLimits } = storeToRefs(useVueTorrentStore())
+const { statusFilter } = storeToRefs(useTorrentStore())
+
+const downloadingStates = [TorrentState.DOWNLOADING, TorrentState.DL_FORCED, TorrentState.FORCED_META_DOWNLOAD]
```
Comment (Larsluph) [inline src/components/Navbar/SideWidgets/CurrentSpeed.vue:16]: I'm not sure about the forced meta state, is it really considered as downloading?

File: src/components/Navbar/SideWidgets/CurrentSpeed.vue [hunk 4]
```diff
@@ -1,12 +1,40 @@
...
+  statusFilter.value = uploadingStates
+}
+
+const isDownloadingFilterActive = computed(
+  () => downloadingStates.every(state => statusFilter.value.includes(state)) && downloadingStates.length == statusFilter.value.length
+)
+
+const isUploadingFilterActive = computed(
+  () => statusFilter.value.every(state => uploadingStates.includes(state)) && uploadingStates.length == statusFilter.value.length
+)
```
Comment (Larsluph) [inline src/components/Navbar/SideWidgets/CurrentSpeed.vue:37]: Use type assertion as well

```suggestion
const isDownloadingFilterActive = computed(
  () => downloadingStates.every(state => statusFilter.value.includes(state)) && downloadingStates.length === statusFilter.value.length
)

const isUploadingFilterActive = computed(
  () => statusFilter.value.every(state => uploadingStates.includes(state)) && uploadingStates.length === statusFilter.value.length
)
```

**Review by Larsluph — CHANGES_REQUESTED** [review]
Great addition!

File: src/components/Navbar/SideWidgets/CurrentSpeed.vue (hunk 3 above)
Comment (jagadam97) [reply src/components/Navbar/SideWidgets/CurrentSpeed.vue:16]: mostly it means used has clicked on force resume and downloaoding state is metadata downloading. 

I will remove this Just for consistency

**Review by Larsluph — APPROVED** [review]
@jagadam97 I merged both SpeedCard and SpeedCardClickable components to prevent a lot of duplication. Apart from this everything looks great, thanks again for the contribution!

**Pull request merged**